src/Swarm/Game/State.hs

```diff
@@ -686,14 +693,22 @@ messageNotifications = to getNotif
 messageIsRecent :: GameState -> LogEntry -> Bool
 messageIsRecent gs e = addTicks 1 (e ^. leTime) >= gs ^. ticks
 
-messageIsFromNearby :: Location -> LogEntry -> Bool
-messageIsFromNearby l e = manhattan l (e ^. leLocation) <= hearingDistance
+-- | Reconciles the possibilities of log messages being
+-- omnipresent and robots being in different worlds
+messageIsFromNearby :: Cosmo Location -> LogEntry -> Bool
+messageIsFromNearby l e = case e ^. leLocation of
+  Omnipresent -> True
+  Located x -> f x
+ where
+  f logLoc = case cosmoMeasure manhattan l logLoc of
+    InfinitelyFar -> False
+    Measurable x -> x <= hearingDistance
```

Comment:
I clarified the semantics of log message proximity.  The code had been suffering from ["Maybe blindness"](https://www.reddit.com/r/haskell/comments/8eecyu/comment/dxuqb33/?utm_source=share&utm_medium=web2x&context=3) IMO, so I created two new isomorphic types:
* `DistanceMeasure`
* `LogLocation`